python/deptry/dependency_getter/builder.py

```diff
@@ -45,21 +45,89 @@ def build(self) -> DependencyGetter:
                 return PoetryDependencyGetter(self.config, self.package_module_name_map)
 
             if self._project_uses_pdm(pyproject_toml):
-                return PDMDependencyGetter(self.config, self.package_module_name_map, self.pep621_dev_dependency_groups)
+                return PDMDependencyGetter(
+                    self.config,
+                    self.package_module_name_map,
+                    self.pep621_dev_dependency_groups,
+                )
+
+            if self._project_uses_setuptools(pyproject_toml) and self._project_uses_dynamic_dependencies(
+                pyproject_toml
+            ):
+                requirements_txt_file = self._project_dynamic_requirements_file(pyproject_toml)
+                return RequirementsTxtDependencyGetter(
+                    self.config, self.package_module_name_map, requirements_txt_file, ()
+                )
 
             if self._project_uses_pep_621(pyproject_toml):
                 return PEP621DependencyGetter(
-                    self.config, self.package_module_name_map, self.pep621_dev_dependency_groups
+                    self.config,
+                    self.package_module_name_map,
+                    self.pep621_dev_dependency_groups,
                 )
 
         check, requirements_files = self._project_uses_requirements_files()
         if check:
             return RequirementsTxtDependencyGetter(
-                self.config, self.package_module_name_map, requirements_files, self.requirements_files_dev
+                self.config,
+                self.package_module_name_map,
+                requirements_files,
+                self.requirements_files_dev,
             )
 
         raise DependencySpecificationNotFoundError(self.requirements_files)
 
+    def _project_uses_setuptools(self, pyproject_toml: dict[str, Any]) -> bool:
```

Comment:
Could you move the 3 new methods between `_project_uses_pdm` and `_project_uses_pep_621`? That way we use the same order as the order methods are called from `build` method.